### PR TITLE
Rename ExtendedTransformer to TransformerNC

### PR DESCRIPTION
**Pull request by JeppeBlixen**
Rename ExtendedTransformer to TransformerNC
Rename TimeZoneTransformer to TimeZoneTransformerNC
Update tests and docs

### CODE REVIEW

**Review by RadekBuczkowski — APPROVED** [review]

**Review by RadekBuczkowski — APPROVED** [review]

File: src/atc/transformers/timezone_transformer.py [hunk 1]
```diff
@@ -3,11 +3,11 @@
 from timezonefinder import TimezoneFinder
```
Comment (frederikgjensen) [inline src/atc/transformers/timezone_transformer.py:4]: maybe rename the file to "timezone_transformer_nc.py"

File: src/atc/transformers/__init__.py [hunk 2]
```diff
@@ -2,5 +2,5 @@
     DropOldestDuplicatesTransformer,
 )
 from .simple_sql_transformer import SimpleSqlServerTransformer  # noqa: F401
-from .timezone_transformer import TimeZoneTransformer  # noqa: F401
+from .timezone_transformer import TimeZoneTransformerNC  # noqa: F401
```
Comment (frederikgjensen) [inline src/atc/transformers/__init__.py:5]: if you rename, then remember to adjust the import

File: tests/local/transformers/test_timezone_transformer_nc.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 from pyspark.sql import types as T
 
 from atc.spark import Spark
-from atc.transformers import TimeZoneTransformer
+from atc.transformers import TimeZoneTransformerNC
```
Comment (frederikgjensen) [inline tests/local/transformers/test_timezone_transformer_nc.py:5]: remember to adjust import here if you change the name of the file

File: tests/local/transformers/test_timezone_transformer_nc.py [hunk 4]
```diff
@@ -2,7 +2,7 @@
...
 
 
 class TimeZoneTransformerTest(DataframeTestCase):
```
Comment (frederikgjensen) [inline tests/local/transformers/test_timezone_transformer_nc.py:8]: maybe rename to "TimeZoneTransformerNCTest" to be consistent

File: examples/etl/etl_nc_example.py [hunk 5]
```diff
@@ -48,12 +48,12 @@ def read(self) -> DataFrame:
         )
 
 
-class IntegerTransformer(ExtendedTransformer):
+class IntegerTransformer(TransformerNC):
```
Comment (frederikgjensen) [inline examples/etl/etl_nc_example.py:51]: maybe rename class names to "...NC" to be consistent

File: docs/etl/README.template.md [hunk 6]
```diff
@@ -167,8 +167,17 @@ It is important that the first transformer is a `MultiInputTransformer` when hav
 
 ### Example-7
 
+Using the TransformerNC(Non Consuming), a transformation step can use dataframes in the dataset from all previous stages.
```
Comment (frederikgjensen) [inline docs/etl/README.template.md:170]: do you mean - "a transformation step can use dataframes in the dataset GROUP"?

**Review by frederikgjensen — COMMENTED** [review]
I have added some minor comments

File: docs/etl/README.template.md [hunk 7]
```diff
@@ -167,8 +167,17 @@ It is important that the first transformer is a `MultiInputTransformer` when hav
 
 ### Example-7
 
+Using the TransformerNC(Non Consuming), a transformation step can use dataframes from all previous stages.
+
+```
+{etl_nc_example}
```
Comment (LauJohansson) [inline docs/etl/README.template.md:173]: is there a example coming up?

File: src/atc/etl/__init__.py [hunk 8]
```diff
@@ -12,7 +12,7 @@
     "Transformer",
     "Orchestrator",
     "EtlBase",
-    "ExtendedTransformer",
+    "TransformerNC",
     "ExtendedLoader",
```
Comment (LauJohansson) [inline src/atc/etl/__init__.py:16]: ExtendedLoader. Shoud it be renamed to LoaderNC too?

File: src/atc/etl/__init__.py [hunk 9]
```diff
@@ -1,9 +1,9 @@
 from .extended_loader import ExtendedLoader
```
Comment (LauJohansson) [inline src/atc/etl/__init__.py:1]: extendedloader -> LoaderNC

File: src/atc/etl/__init__.py (hunk 8 above)
Comment (LauJohansson) [reply src/atc/etl/__init__.py:16]: Do it in a new PR.

File: src/atc/etl/__init__.py (hunk 9 above)
Comment (LauJohansson) [reply src/atc/etl/__init__.py:1]: Do it in a new PR.

**Review by LauJohansson — APPROVED** [review]

**Pull request merged**